### DISCUSSION TITLE
Why face box is includes neck but not hairs?

### DISCUSSION THREAD

**Discussion by enszrlu** [vladmandic/human#419] [Q&A]
I have been experimenting the library for last couple days. I am trying to use face box to extract face from the photos. But I believe intention for face box is to draw a box around the face(not including hair) with a little bit buffer. Most of the time it includes neck and little bit of hair but when hair is longer, it does not capture entire hair. Is there a way to include hairs as well? I was planning  maybe deleting the background and scanning the photo from top to bottom around face box, first point would be top of the hair, then do the same on left and right. 

Please see photo below as example. 

![response3](https://github.com/vladmandic/human/assets/65814876/57fa988c-74c6-4a7d-9cc0-284e60704f22)

![response4](https://github.com/vladmandic/human/assets/65814876/6eb3e580-c3a7-4fd2-85da-89024e475544)

**Comment by vladmandic** [comment]
you can see the full matrix of detection points - as you can see, there are no detection points in the areas outside of the face.
and box is simply an expanded rectangle looking at min/max of all detected points. you can control how much its expanded in the config.

**Reply by enszrlu** [reply]
I thought so, thanks for response. So there is no built in functionality to detect hair as well?

**Reply by vladmandic** [reply]
no, sorry.
but if your goal is to remove background, i suggest to use some image segmentation model - for example, https://github.com/xuebinqin/DIS is pretty good and not too large.

**Reply by enszrlu** [reply]
Thanks for the suggestion. I am able to remove the background with another library but I will look into this one as well.

**Comment by enszrlu** [comment]
> you can see the full matrix of detection points - as you can see, there are no detection points in the areas outside of the face. and box is simply an expanded rectangle looking at min/max of all detected points. you can control how much its expanded in the config.

I was looking thru docs to find how I can change how much its expanded but I could not find it. Can you please point me to it? Thanks

**Reply by vladmandic** [reply]
my mistake, its not user-configurable, right now its hard-coded in:

https://github.com/vladmandic/human/blob/c55279ca8207134802453c12c318f28128750362/src/face/faceboxes.ts#L18

if needed, i can expose it as configurable value.

**Reply by enszrlu** [reply]
That would be great. I think it would be useful to be able to change it.

**Reply by vladmandic** [reply]
ok, i'll do a minor release next week.

**Reply by vladmandic** [reply]
release 3.2.1 will be out shortly, new options are `config.face.detector.scale` and `config.face.iris.scale`

**Reply by enszrlu** [reply]
tested it now. I am not sure if it works fine or not. I am getting weird results, maybe it is not working well with draw function now. But also I am not sure what scale should be for no enlargement. (I thought scale=1 should work but it is slightly bigger than no_scale polylines.) See below photos please. 

```
  const humanConfig = {
    // backend: 'tensorflow',
    modelBasePath: "file://node_modules/@vladmandic/human/models",
    debug: false,
    face: { enabled: true, detector: { maxDetected: 10, scale: 1 } }, 
    hand: {
      enabled: true,
      maxDetected: 20,
      minConfidence: 0.5,
      detector: { modelPath: "handtrack.json" },
    }, // use alternative hand model
    body: { enabled: true },
    object: { enabled: true },
    gestures: { enabled: true },
  };
```

**Scale = 0.7**
![scale07](https://github.com/vladmandic/human/assets/65814876/65b1c630-2f2a-448e-a41f-8b108b4e84ca)

**No Scale** 
![no_scale](https://github.com/vladmandic/human/assets/65814876/c6f6c37c-47f5-4aae-8527-89fd1f24f334)

**Scale = 1**
![scale1](https://github.com/vladmandic/human/assets/65814876/da1f10f7-3cc0-4c46-a448-c78156ccb9bb)

**Scale = 1.8**
![scale18](https://github.com/vladmandic/human/assets/65814876/39b3ff9f-520d-4b26-96fa-c42bfbded516)
